### PR TITLE
add accessory dao service

### PR DESCRIPTION
**Pull request by wy65701436**
Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

**Review by chlins — APPROVED** [review]
/lgtm

File: make/migrations/postgresql/0080_2.5.0_schema.up.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* create table of accessory */
+CREATE TABLE IF NOT EXISTS artifact_accessory (
+    id SERIAL PRIMARY KEY NOT NULL,
+    /*
+       the artifact id of the accessory itself.
+    */
+    artifact_id int,
```
Comment (heww) [inline make/migrations/postgresql/0080_2.5.0_schema.up.sql:7]: bigint?

File: make/migrations/postgresql/0080_2.5.0_schema.up.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    /*
+     the subject artifact id of the accessory.
+    */
+    subject_artifact_id int,
```
Comment (heww) [inline make/migrations/postgresql/0080_2.5.0_schema.up.sql:11]: bigint?

File: src/pkg/accessory/model/accessory.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	RefProvider
+	RefIdentifier
+	GetID() int64
+	GetArtifactID() int64
+	GetSubjectArtID() int64
+	GetType() string
+	GetSize() int64
+	GetDigest() string
+	GetCreateTime() time.Time
+	GetIcon() string
```
Comment (heww) [inline src/pkg/accessory/model/accessory.go:75]: It's crazy for these methods in this interface here.

```go
type struct AccessoryData {
    ID int64
    ArtifactID int64 
   ....
}

type Accessory interface {
        RefProvider
	RefIdentifier
        GetAccessoryData() AccessoryData
}
```

File: src/pkg/accessory/model/options.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+package model
+
+import "time"
+
+// Option function to set the options of the cache
+type Option func(*Options)
+
+// Options options of the cache
+type Options struct {
```
Comment (heww) [inline src/pkg/accessory/model/options.go:23]: If there is a method `GetAccessoryData() AccessoryData` in the interface, this `Options` is not needed.

File: src/pkg/accessory/model/accessory.go (hunk 3 above)
Comment (heww) [reply src/pkg/accessory/model/accessory.go:75]: BTW, if there is only one implementation of this interface at the moment, I don't think this interface is needed.

Comment (wy65701436) [reply src/pkg/accessory/model/accessory.go]: for futher support, it will have SBom, Nydus.

**Review by heww — APPROVED** [review]

**Pull request merged**